scripts/characters/constructor.gd

```diff
@@ -701,20 +727,30 @@ func setRace(person,mother):
 	
 	globals.traceFile('Constructor.pickRace Completed')
 
-func setRaceDisplay(person):
+func setRaceDisplay(person): #ralphD - reworked to fix bugs
 	var text = ""
 	var primaryrace = ""
 	var primaryracepercent = 0
 	var secondaryrace = ""
 	var secondaryracepercent = 0
+	var racetiestobreak = [] #ralphD
 	
 	#Sort Races
 	for race in genealogies:
 		if person.genealogy[race] > 0 && person.genealogy[race] >= primaryracepercent:
 			#Push Primary to Secondary if Greater
-			if primaryracepercent > secondaryracepercent:
+			if person.genealogy[race] > primaryracepercent:
 				secondaryrace = primaryrace
 				secondaryracepercent = primaryracepercent
+				primaryrace = race
+				primaryracepercent = person.genealogy[race]
+				racetiestobreak = []
+			#ralphD
+			elif person.genealogy[race] == primaryracepercent:
+				racetiestobreak.append_array([race])
```

Comment:
there is no reason to use append_array() over append() here, it's just inefficient

Reply:
claw hammer?  well that is a lot better than the sledge I found somewhere else, I'll give that a try.  ;]

Reply:
comment incorporated in new commit